internal/services/loganalytics/log_analytics_workspace_table_resource.go

```diff
@@ -19,13 +20,32 @@ type LogAnalyticsWorkspaceTableResource struct {
 }
 
 var _ sdk.ResourceWithUpdate = LogAnalyticsWorkspaceTableResource{}
+var _ sdk.ResourceWithCustomizeDiff = LogAnalyticsWorkspaceTableResource{}
 
 type LogAnalyticsWorkspaceTableResourceModel struct {
 	Name            string `tfschema:"name"`
 	WorkspaceId     string `tfschema:"workspace_id"`
+	Plan            string `tfschema:"plan"`
 	RetentionInDays int64  `tfschema:"retention_in_days"`
 }
 
+func (r LogAnalyticsWorkspaceTableResource) CustomizeDiff() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			rd := metadata.ResourceDiff
+
+			if string(tables.TablePlanEnumBasic) == rd.Get("plan").(string) {
+				if _, ok := rd.GetOk("retention_in_days"); ok {
+					return fmt.Errorf("cannot set retention_in_days as it is immutabl on Basic plan")
```

Comment:
Instead of "immutable" should we use the wording is "Retention is fixed at eight days".

Reply:
@zioproto thanks for your feedback.  I'm using `immutable ` because I'm not sure if "8 days" would change to something else in the future. If changes might occur, TF may not be able to modify it in time, thus giving users wrong prompts. Of course, I'd be happy to put "8 days" in the message if the service team could confirm that this number won't change in the future. Could you help confirm this? Thanks in advance.

Reply:
The context is that the error messages should give information that helps the customer to find the documentation searching on the web.

The problem here is that a customer searching on Google "retention_in_days as it is immutable on Basic Plan" will not find anything in the Azure documentation. Instead searching "retention_in_days fixed at 8 days" will return the correct Azure documentation page where this is explained.

Reply:
Thanks for the explanation. The code has been updated.